cmd/subctl/recover_broker_info.go

```diff
@@ -63,114 +55,57 @@ func init() {
+	status.Success("Found Broker installed on a different cluster in namespace %s", brokerNamespace)
+
 	//nolint:wrapcheck // No need to wrap errors here.
-	return recoverRestConfigProducer.RunOnSelectedContext(
-		func(submCluster *cluster.Info, namespace string, status reporter.Interface) error {
-			if isSubmJoinedToBroker(clusters, submCluster) {
-				status.Success("Found a Submariner installation on cluster %q joined to the Broker", submCluster.Name)
-				//nolint:wrapcheck // No need to wrap errors here.
-				return broker.RecoverData(brokerCluster, submCluster, brokerObj, namespace, status)
-			}
-
-			return status.Error(
-				fmt.Errorf("submariner is not installed on cluster %s. "+
-					"Please specify the cluster where Submariner is installed via `--kubeconfig` or `--context` flag"+
-					"", submCluster.Name), "")
-		}, status)
+	return broker.RecoverData(submCluster, brokerObj, brokerNamespace, brokerRestConfig, status)
 }
 
-func recoverBrokerFromCurrentContext(clusterInfo *cluster.Info, namespace string, status reporter.Interface) error {
-	// if --brokerconfig not provided, search for broker on current context and proceed to get Submariner
-	brokerObj, err := getBroker(clusterInfo, namespace, "Please specify the cluster where the Broker is installed "+
-		"via the `--brokerconfig` or `--brokercontext` flag and the namespace via the `--brokernamespace` flag", status)
-	if err != nil {
-		return err
-	}
-
-	clusters, err := clusterInfo.GetClusters(namespace)
+func getBroker(config *rest.Config, namespace string) (*v1alpha1.Broker, error) {
+	brokerClientProducer, err := client.NewProducerFromRestConfig(config)
 	if err != nil {
-		return status.Error(err, "error listing joined clusters")
-	}
-
-	ok, err := tryToRecoverFromBroker(clusterInfo, brokerObj, namespace, clusters, status)
-	if ok || err != nil {
-		return err
+		return nil, errors.Wrap(err, "Error creating broker client Producer")
 	}
 
-	return status.Error(
-		fmt.Errorf("submariner is not installed on cluster %q. "+
-			"Please specify the cluster where Submariner is installed via `--kubeconfig` or `--context` flag"+
-			"", clusterInfo.Name), "")
-}
+	brokerObj := &v1alpha1.Broker{}
+	err = brokerClientProducer.ForGeneral().Get(
+		context.TODO(), controllerClient.ObjectKey{
+			Namespace: namespace,
+			Name:      brokercr.Name,
+		}, brokerObj)
 
-func getBroker(clusterInfo *cluster.Info, namespace, notFoundMsg string, status reporter.Interface) (*v1alpha1.Broker, error) {
-	status.Start("Checking if the Broker is installed on cluster %q in namespace %q", clusterInfo.Name, namespace)
-
-	brokerObj, err := clusterInfo.GetBroker(namespace)
 	if apierrors.IsNotFound(err) {
-		return nil, status.Error(fmt.Errorf("the Broker is not installed on the specified cluster in namespace %s. %s", notFoundMsg,
-			namespace), "")
+		return nil, errors.Wrapf(err, "the Broker is not installed on the specified cluster in namespace %s", namespace)
```

Comment:
This is not an error condition. We should return nil error. The `nilnil` linter may complain about this in which case return a `found` bool.